upstream/reconcile.go

```diff
@@ -76,7 +81,6 @@ func (t *upstreamReconciler) sync(ctx context.Context, table, next string) error
 		if err != nil {
 			return fmt.Errorf("failed to fetch hash of primary keys from local db: %w", err)
 		}
-		next = localStatus.Next
```

Comment:
Don't we need this in here though instead of inside `if upstreamStatus.Hash == localStatus.Hash`?

The pagination would be stuck in a loop otherwise.

Reply:
We can add a counter maybe ? currently we do not check the hash after adding missing resources